packages/frontend/src/components/VisualizationConfigs/ChartConfigPanel/CustomVisConfigTabs.tsx

```diff
@@ -65,6 +157,257 @@ const loadMonaco = (monaco: Monaco, schemas: Schema[]) => {
     });
 };
 
+const registerCustomCompletionProvider = (
+    monaco: Monaco,
+    language: string,
+    fields: string[],
+) => {
+    console.debug('Loading completion provider with fields', fields);
+    return monaco.languages.registerCompletionItemProvider(language, {
+        provideCompletionItems: (model, position) => {
+            const wordUntilPosition = model.getWordUntilPosition(position);
+            const range = {
+                startLineNumber: position.lineNumber,
+                endLineNumber: position.lineNumber,
+                startColumn: wordUntilPosition.startColumn,
+                endColumn: wordUntilPosition.endColumn,
+            };
+
+            const suggestions: languages.CompletionItem[] = fields.map(
+                (field) => {
+                    return {
+                        label: field,
+                        kind: monaco.languages.CompletionItemKind.Class,
+                        insertText: field,
+                        range,
+                    };
+                },
+            );
+
+            return { suggestions };
+        },
+        triggerCharacters: ['$'],
+    });
+};
+const SelectTemplate = ({
+    itemsMap,
+    isCustomConfig,
+    // isEditorEmpty,
+    setEditorConfig,
+}: {
+    itemsMap: ItemsMap | undefined;
+    isCustomConfig: boolean;
+    isEditorEmpty: boolean;
+    setEditorConfig: (config: string) => void;
+}) => {
+    /*const [selectedTemplate, setSelectedTemplate] = useState<
```

Comment:
is this needed?

Reply:
This was commented out to support a confirm dialog, but if we don't need it, I can remove it.

Reply:
thanks 🙌